spinta/backends/__init__.py

```diff
@@ -229,6 +231,43 @@ def simple_data_check(
         simple_data_check(context, data, dtype, prop, dtype.backend, v)
 
 
+@commands.simple_data_check.register(Context, DataItem, ExternalRef, Property, Backend, dict)
+def simple_data_check(
+    context: Context,
+    data: DataItem,
+    dtype: ExternalRef,
+    prop: Property,
+    backend: Backend,
+    value: dict,
+) -> None:
+    denorm_prop_keys = [
+        p.name.split('.', maxsplit=1)[-1]
+        for p in prop.model.properties.values() if (
+            isinstance(p.dtype, Denorm) and
+            p.name.split('.')[0] == prop.name
+        )
+    ]
+    allowed_keys = [prop.name for prop in dtype.refprops]
+    allowed_keys.extend(denorm_prop_keys)
+    value = flatten_value(value)
+    for key in value.keys():
+        if key not in allowed_keys:
+            raise exceptions.FieldNotInResource(prop, property=key)
+
+
+@commands.simple_data_check.register(Context, DataItem, Ref, Property, Backend, object)
+def simple_data_check(
+    context: Context,
+    data: DataItem,
+    dtype: Ref,
+    prop: Property,
+    backend: Backend,
+    value: object,
+) -> None:
+    if value and not isinstance(value, (dict, str)):
```

Comment:
Why `str` is allowed too?

Reply:
The test that fails without str is [`test_manifest_loading`](https://github.com/atviriduomenys/spinta/blob/95dbf1c2949b995deae2a7141453fad57a006bc8/tests/test_manifests.py#L28). From what I understand, during `cli.invoke(rc, ['migrate'])`, [`update_schema_version`](https://github.com/atviriduomenys/spinta/blob/95dbf1c2949b995deae2a7141453fad57a006bc8/spinta/manifests/backend/helpers.py#L232) method is called and `version` attribute is passed as string.

Reply:
In this case we definately should fix the `update_schema_version` and here we should expect `value` to be `dict`.